### PR TITLE
Improved the seeding and usage of rand

### PR DESCRIPTION
**Pull request by aviaconstructor**
## Description

Seeding `rand` each time before its use is slow, and it leads to lower quality of randomness. Improved the seeding by moving it out to `func init()`. After the change, since the seeding is done in many modules independently, it will be performed multiple times at the startup.  After moving to go 1.20 the `init()` funcs related to rand seeding can be removed, see https://pkg.go.dev/math/rand#Seed 

Added the test `TestRandomStrElement` to demonstrate how seeding with `time.Now().Unix()` can lead to generation of the same value 100 times in a row. The proposed fix makes the test succeed.

TODO: There is an open issue with `func handleRolloutEpoch`. It seems, the result `controlPlaneManager.internal.rolloutEpoch` will be a random int64, negative in about 1/2 of cases. If this is an intent, it can be implemented in a much less convoluted way)))

## What is the expected result?

Higher quality of insecure randomness, slightly better performance. The newly added test will not fail

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: dhctl, modules
type: fix
summary: Improved the seeding and usage of rand
impact: Higher quality of insecure randomness, slightly better performance
impact_level: low
```

### CODE REVIEW

**Review by shvgn — COMMENTED** [review]
I see no issue with it, but it must be reviewed by other code owners. Also, a copyright year should be fixed

File: dhctl/pkg/util/stringsutil/strings_test.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
+// Copyright 2021 Flant JSC
```
[the remark this answers is not in the document]
Comment (aviaconstructor) [reply dhctl/pkg/util/stringsutil/strings_test.go]: Done

**Comment by aviaconstructor** [comment]
A friendly reminder on this PR - the copyright year is fixed as requested..

**Comment by aviaconstructor** [comment]
I see this PR is stalled, but maybe for good - in go 1.20 the seeding of rand is done automatically - https://tip.golang.org/doc/go1.20 

If you plan to move to go 1.20/1.21 any time soon please let me know and I remove the seeding. Or just consider the merge now as the seeding as in the PR will be harmless in 1.2x

File: modules/040-control-plane-manager/hooks/rollout_epoch.go [hunk 2]
```diff
@@ -36,13 +36,18 @@ var _ = sdk.RegisterFunc(&go_hook.HookConfig{
...
-	var seed = binary.BigEndian.Uint64([]byte(clusterUUID))
-	rand.Seed(int64(seed))
+	seed := binary.BigEndian.Uint64([]byte(clusterUUID))
+	randomSource := rand.NewSource(int64(seed))
+	random := rand.New(randomSource) //nolint:gosec
 
-	epoch := ((rand.Int63() * 864000) + time.Now().Unix()) / 864000
+	// TODO ?? could not understand what is going on here as the result epoch can even be negative.
+	// Most likely this whole function has to be rewritten
+	epoch := ((random.Int63() * tenDaysInSeconds) + time.Now().Unix()) / tenDaysInSeconds
```
Comment (yalosev) [inline modules/040-control-plane-manager/hooks/rollout_epoch.go:50]:
```suggestion
	epoch := ((int64(random.Uint32()) * tenDaysInSeconds) + time.Now().Unix()) / tenDaysInSeconds
```

**Review by yalosev — DISMISSED** [review]

**Review by yalosev — APPROVED** [review]

**Review by name212 — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by yalosev** [comment]
@aviaconstructor Thank you for the PR. Sorry for such a long review.

**Pull request merged**